### ISSUE TITLE
How many epochs will my model train for?

### ISSUE THREAD

**Issue by avinregmi** [PyTorchLightning/pytorch-lightning#1627]
#### How many epochs will my model train for if i don't set max and min epoch value in my trainer?

trainer = Trainer(gpus=1,max_epochs=4)

I know that I could specify max and min epochs. What if i don't specify and just call fit() without min or max epochs. What value does it use to stop training my model? Is it loss value returned from training_step(). Also will it check if that loss is in a similar range as my validation loss so it does not over or under fit? 

Thanks

**Comment by yukw777** [comment]
The doc is your friend: https://pytorch-lightning.readthedocs.io/en/latest/trainer.html#max-epochs
You’ll see that the default value for max_epochs is 1000.

Also, it sounds like you’re looking for this: https://pytorch-lightning.readthedocs.io/en/latest/callbacks.html#early-stopping

**Comment by avinregmi** [comment]
thanks @yukw777 !! Is there a way to monitor two different quantities in the same call back. What if i want to monitor loss and custom metric 'accuracy' for early stopping?

**Comment by yukw777** [comment]
You’d have to write your own custom callback.

**Comment by avinregmi** [comment]
thanks again!

**Comment by akamboj2** [comment]
looks like site changed. Here is the new doc: https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html#max-epochs
